### PR TITLE
Update `@wordpress/eslint-plugin`

### PR DESCRIPTION
**Pull request by anomiex**
## Proposed changes:

This brings two new rules that needed fixing:

* `@wordpress/i18n-no-flanking-whitespace` complains about leading or trailing whitespace in JS i18n messages. Fix depends on context:
  * Sometimes it makes no difference, the browser (or React) strips it out. Removed these.
  * Sometimes it does, or it's unclear. In these cases I added it back outside the i18n call.
* `@wordpress/i18n-hyphenated-range` wants en-dashes (or em-dashes) for numeric ranges. Switched them to en-dashes.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

None

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Places where these messages are used still looks right?

### CODE REVIEW

File: projects/packages/search/src/dashboard/components/upsell-nudge/index.jsx [hunk 1]
```diff
@@ -14,10 +14,10 @@ export default function InstantSearchUpsellNudge( props = { upgrade: true } ) {
 		<a className="jp-instant-search-upsell-nudge jp-search-dashboard-cut" href={ props.href }>
 			<span>
 				{ __(
-					'Offer instant search results to your visitors as soon as they start typing. ',
+					'Offer instant search results to your visitors as soon as they start typing.',
 					'jetpack-search-pkg'
 				) }
-			</span>
+			</span>{ ' ' }
```
Comment (zinigor) [inline projects/packages/search/src/dashboard/components/upsell-nudge/index.jsx:20]: Should we put the space inside the `span` element to preserve the original behavior?

File: projects/plugins/protect/src/js/components/firewall-footer/index.jsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ const ShareDebugData = () => {
 
 	return (
 		<div className={ styles[ 'share-data-section' ] }>
-			<Title mb={ 2 }>{ __( ' Share detailed data with Jetpack', 'jetpack-protect' ) }</Title>
+			<Title mb={ 2 }>{ __( 'Share detailed data with Jetpack', 'jetpack-protect' ) }</Title>
```
Comment (zinigor) [inline projects/plugins/protect/src/js/components/firewall-footer/index.jsx:73]: Have you been able to figure out why there was a space here in the first place? Maybe we should preserve this one too?

File: projects/plugins/protect/src/js/components/scan-footer/index.jsx [hunk 3]
```diff
@@ -33,7 +33,7 @@ const ProductPromotion = () => {
 				<Title>{ __( 'Get access to our Cloud', 'jetpack-protect' ) }</Title>
 				<Text mb={ 3 }>
 					{ __(
-						'With your Protect upgrade, you have free access to scan your site on our Cloud, so you can be aware and fix your threats even if your site goes down. ',
+						'With your Protect upgrade, you have free access to scan your site on our Cloud, so you can be aware and fix your threats even if your site goes down.',
```
Comment (zinigor) [inline projects/plugins/protect/src/js/components/scan-footer/index.jsx:36]: I'm afraid we'll have to preserve this too for inline elements.

**Review by zinigor — COMMENTED** [review]
I have some questions about preserving spaces here, maybe we should test these to make sure.

File: projects/packages/search/src/dashboard/components/upsell-nudge/index.jsx (hunk 1 above)
Comment (anomiex) [reply projects/packages/search/src/dashboard/components/upsell-nudge/index.jsx:20]: 🤷 No idea. In practice CSS currently sets `display: block` on these spans so the whitespace gets ignored either way.

File: projects/plugins/protect/src/js/components/firewall-footer/index.jsx (hunk 2 above)
Comment (anomiex) [reply projects/plugins/protect/src/js/components/firewall-footer/index.jsx:73]: It seems to go back to 021eb6a5079af049176bf3563b545f764f3d4fef, beyond that I have no idea.

It made no difference to the rendering when I tried it. HTML ignores whitespace at the start of block elements like the `<h2>` generated here.

Before:
before

After:
after

File: projects/plugins/protect/src/js/components/scan-footer/index.jsx (hunk 3 above)
Comment (anomiex) [reply projects/plugins/protect/src/js/components/scan-footer/index.jsx:36]: What inline elements? The following element is a block-formatted button.

Screenshot

File: projects/packages/search/src/dashboard/components/upsell-nudge/index.jsx (hunk 1 above)
Comment (anomiex) [reply projects/packages/search/src/dashboard/components/upsell-nudge/index.jsx:20]: Before:
before

After:
after

**Review by zinigor — APPROVED** [review]
Thank you for doing the legwork and actually testing the changes!

**Pull request merged**